app/services/manifest_validator.rb

```diff
@@ -0,0 +1,28 @@
+class ManifestValidator
+  include JsonSchemaHelper
+
+  attr_reader :content, :errors
+
+  def initialize(content)
+    if content.is_a?(String)
+      @content = JSON.parse(content)
+    else
+      @content = content
+    end
+    register_schemas_by_uri
+  end
+
+  def run
+    schema_file_path = schema_file('post-manifest')
+    @errors = JSON::Validator.fully_validate(schema_file_path, content, errors_as_objects: true)
+    !@errors.any?
```

Comment:
I toyed with these 2 interfaces:

```ruby
# exception-based
begin
  validator.run
rescue ManifestValidationError => error
  puts "You had problems: #{error.messages("\n")}"
end
# or return code-based
unless validator.run
  puts "You had problems: #{validator.error_messages.join("\n")}"
end
```

Both of those seem consistent with how Ruby/Rails handles errors. I kind of prefer the former, since catastrophic errors are your friend, but it seemed less verbose to have `run` return a simple boolean.